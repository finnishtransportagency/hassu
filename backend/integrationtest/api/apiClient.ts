import { AbstractApi, OperationConfig } from "../../../common/abstractApi";
import { AppSyncResolverEvent } from "aws-lambda/trigger/appsync-resolver";
import { AppSyncEventArguments, handleEvent } from "../../src/apiHandler";

class API extends AbstractApi {
  async callAPI(operation: OperationConfig, variables?: unknown): Promise<unknown> {
    return (
      await handleEvent({
        info: { fieldName: operation.name },
        arguments: variables,
      } as AppSyncResolverEvent<AppSyncEventArguments>)
    ).data;
  }

  async callYllapitoAPI(operation: OperationConfig, variables?: unknown): Promise<unknown> {
    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
    const payload = {
      info: { fieldName: operation.name },
      arguments: variables,
    } as AppSyncResolverEvent<AppSyncEventArguments>;
    return (await handleEvent(payload)).data;
  }
}

export const api = new API();
