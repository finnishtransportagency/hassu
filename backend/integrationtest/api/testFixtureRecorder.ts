import fs from "fs";
import { projektiDatabase } from "../../src/database/projektiDatabase";
import { DBProjekti } from "../../src/database/model";
import { DocumentClient } from "aws-sdk/lib/dynamodb/document_client";
import { localDocumentClient } from "../util/databaseUtil";

export enum FixtureName {
  NAHTAVILLAOLO = "NAHTAVILLAOLO",
}

export async function recordProjektiTestFixture(fixtureName: string | FixtureName, oid: string): Promise<void> {
  const dbProjekti = await projektiDatabase.loadProjektiByOid(oid);
  replaceFieldsByName(dbProjekti, "2020-01-01T00:00:00+03:00", "tuotu", "paivitetty");
  replaceFieldsByName(dbProjekti, "salt123", "salt");

  const oldValue = readRecord(fixtureName);
  const currentValue = JSON.stringify(dbProjekti, null, 2);
  // Prevent updating file timestamp so that running tests with "watch" don't get into infinite loop
  if (!oldValue || oldValue !== currentValue) {
    fs.writeFileSync(__dirname + "/records/" + fixtureName + ".json", currentValue);
  }
}

function readRecord(fixtureName: string | FixtureName) {
  const buffer = fs.readFileSync(__dirname + "/records/" + fixtureName + ".json");
  return buffer.toString("utf-8");
}

export async function useProjektiTestFixture(fixtureName: string | FixtureName): Promise<string> {
  const dbProjekti: Partial<DBProjekti> = JSON.parse(readRecord(fixtureName));
  // Write the copied projekti to archive table
  const putParams: DocumentClient.PutItemInput = {
    TableName: process.env.TABLE_PROJEKTI,
    Item: dbProjekti,
  };
  await localDocumentClient.put(putParams).promise();

  return dbProjekti.oid;
}

export function replaceFieldsByName(obj: unknown, value: unknown, ...fieldNames: string[]): void {
  Object.keys(obj).forEach(function (prop) {
    if (typeof obj[prop] == "object" && obj[prop] !== null) {
      replaceFieldsByName(obj[prop], value, ...fieldNames);
    } else if (fieldNames.indexOf(prop) >= 0) {
      obj[prop] = value;
    }
  });
}
