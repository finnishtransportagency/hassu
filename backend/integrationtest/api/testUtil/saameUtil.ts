import { Status } from "../../../../common/graphql/apiModel";
import { ProjektiFixture } from "../../../test/fixture/projektiFixture";
import { deleteProjekti } from "./tests";
import { addLogoFilesToProjekti } from "./util";
import { DBProjekti } from "../../../src/database/model";
import { projektiDatabase } from "../../../src/database/projektiDatabase";

function deleteParts(dbProjekti: DBProjekti, vaihe: Status) {
  delete dbProjekti.jatkoPaatos2VaiheJulkaisut;
  if (vaihe == Status.JATKOPAATOS_2) {
    return;
  }
  delete dbProjekti.jatkoPaatos2Vaihe;
  delete dbProjekti.jatkoPaatos1VaiheJulkaisut;
  if (vaihe == Status.JATKOPAATOS_1) {
    return;
  }
  delete dbProjekti.hyvaksymisPaatosVaiheJulkaisut;
  delete dbProjekti.jatkoPaatos1Vaihe;
  if (vaihe == Status.HYVAKSYMISMENETTELYSSA) {
    return;
  }
  delete dbProjekti.hyvaksymisPaatosVaihe;
  delete dbProjekti.nahtavillaoloVaiheJulkaisut;
  if (vaihe == Status.NAHTAVILLAOLO) {
    return;
  }
  delete dbProjekti.nahtavillaoloVaihe;
  delete dbProjekti.vuorovaikutusKierrosJulkaisut;
  if (vaihe == Status.SUUNNITTELU) {
    return;
  }
  delete dbProjekti.vuorovaikutusKierros;
  delete dbProjekti.aloitusKuulutusJulkaisut;
}

export async function createSaameProjektiToVaihe(vaihe: Status): Promise<DBProjekti> {
  const dbProjekti = new ProjektiFixture().dbProjektiKaikkiVaiheetSaame();
  deleteParts(dbProjekti, vaihe);

  const oid = dbProjekti.oid;
  await deleteProjekti(oid);
  await addLogoFilesToProjekti(oid);
  await projektiDatabase.createProjekti(dbProjekti);

  return dbProjekti;
}
