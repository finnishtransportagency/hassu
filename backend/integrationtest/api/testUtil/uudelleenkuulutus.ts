import { api } from "../apiClient";
import {
  KuulutusJulkaisuTila,
  NykyinenKayttaja,
  Projekti,
  ProjektiJulkinen,
  ProjektiKayttaja,
  Status,
  TilasiirtymaToiminto,
  TilasiirtymaTyyppi,
} from "../../../../common/graphql/apiModel";

import { expectToMatchSnapshot } from "./util";
import { asetaAika, loadProjektiFromDatabase, testPublicAccessToProjekti } from "./tests";
import { UserFixture } from "../../../test/fixture/userFixture";
import {
  cleanupAloituskuulutusTimestamps,
  cleanupHyvaksymisPaatosVaiheJulkaisuJulkinenTimestamps,
  cleanupHyvaksymisPaatosVaiheTimestamps,
  cleanupNahtavillaoloJulkaisuJulkinenTimestamps,
  cleanupNahtavillaoloTimestamps,
} from "./cleanUpFunctions";
import { dateToString, parseDate } from "../../../src/util/dateUtil";

const { expect } = require("chai"); //

export enum UudelleelleenkuulutettavaVaihe {
  ALOITUSKUULUTUS,
  NAHTAVILLAOLO,
  HYVAKSYMISPAATOSVAIHE,
  JATKOPAATOS_1,
  JATKOPAATOS_2,
}

type ProjektinSiivoaja = (projekti: Projekti) => Partial<Projekti>;
type ProjektiJulkinenSiivoaja = (projekti: ProjektiJulkinen) => Partial<ProjektiJulkinen>;
type JulkaisuKey = keyof Pick<
  Projekti,
  | "aloitusKuulutusJulkaisu"
  | "nahtavillaoloVaiheJulkaisu"
  | "hyvaksymisPaatosVaiheJulkaisu"
  | "jatkoPaatos1VaiheJulkaisu"
  | "jatkoPaatos2VaiheJulkaisu"
>;

type LuonnosKey = keyof Pick<
  Projekti,
  "aloitusKuulutus" | "nahtavillaoloVaihe" | "hyvaksymisPaatosVaihe" | "jatkoPaatos1Vaihe" | "jatkoPaatos2Vaihe"
>;

export type UudelleelleenkuulutettavaVaiheenTiedot = {
  initialStatus: Status;
  status: Status;
  publicStatus: Status;
  tilasiirtymaTyyppi: TilasiirtymaTyyppi;
  julkaisuKey: JulkaisuKey;
  luonnosKey: LuonnosKey;
  luonnoksenSiivoaja: ProjektinSiivoaja;
  julkaisunSiivoaja: ProjektinSiivoaja;
  julkaisuJulkinenSiivoaja: ProjektiJulkinenSiivoaja;
};

const uudelleenKuulutusTiedotVaiheelle: Record<UudelleelleenkuulutettavaVaihe, UudelleelleenkuulutettavaVaiheenTiedot> = {
  [UudelleelleenkuulutettavaVaihe.ALOITUSKUULUTUS]: {
    initialStatus: Status.SUUNNITTELU,
    status: Status.SUUNNITTELU,
    publicStatus: Status.ALOITUSKUULUTUS,
    tilasiirtymaTyyppi: TilasiirtymaTyyppi.ALOITUSKUULUTUS,
    julkaisuKey: "aloitusKuulutusJulkaisu",
    luonnosKey: "aloitusKuulutus",
    luonnoksenSiivoaja: ({ aloitusKuulutus }) => ({
      aloitusKuulutus: cleanupAloituskuulutusTimestamps(aloitusKuulutus) as Projekti["aloitusKuulutus"],
    }),
    julkaisunSiivoaja: ({ aloitusKuulutusJulkaisu }) => ({
      aloitusKuulutusJulkaisu: cleanupAloituskuulutusTimestamps(aloitusKuulutusJulkaisu) as Projekti["aloitusKuulutusJulkaisu"],
    }),
    julkaisuJulkinenSiivoaja: ({ aloitusKuulutusJulkaisu }) => ({
      aloitusKuulutusJulkaisu,
    }),
  },
  [UudelleelleenkuulutettavaVaihe.NAHTAVILLAOLO]: {
    initialStatus: Status.NAHTAVILLAOLO,
    status: Status.NAHTAVILLAOLO,
    tilasiirtymaTyyppi: TilasiirtymaTyyppi.NAHTAVILLAOLO,
    publicStatus: Status.NAHTAVILLAOLO,
    julkaisuKey: "nahtavillaoloVaiheJulkaisu",
    luonnosKey: "nahtavillaoloVaihe",
    luonnoksenSiivoaja: ({ nahtavillaoloVaihe }) => ({
      nahtavillaoloVaihe: cleanupNahtavillaoloTimestamps(nahtavillaoloVaihe) as Projekti["nahtavillaoloVaihe"],
    }),
    julkaisunSiivoaja: ({ nahtavillaoloVaiheJulkaisu }) => ({
      nahtavillaoloVaiheJulkaisu: cleanupNahtavillaoloTimestamps(nahtavillaoloVaiheJulkaisu) as Projekti["nahtavillaoloVaiheJulkaisu"],
    }),
    julkaisuJulkinenSiivoaja: ({ nahtavillaoloVaihe }) => ({
      nahtavillaoloVaihe: cleanupNahtavillaoloJulkaisuJulkinenTimestamps(nahtavillaoloVaihe),
    }),
  },
  [UudelleelleenkuulutettavaVaihe.HYVAKSYMISPAATOSVAIHE]: {
    initialStatus: Status.HYVAKSYTTY,
    status: Status.HYVAKSYTTY,
    publicStatus: Status.HYVAKSYTTY,
    tilasiirtymaTyyppi: TilasiirtymaTyyppi.HYVAKSYMISPAATOSVAIHE,
    julkaisuKey: "hyvaksymisPaatosVaiheJulkaisu",
    luonnosKey: "hyvaksymisPaatosVaihe",
    luonnoksenSiivoaja: ({ hyvaksymisPaatosVaihe }) => ({
      hyvaksymisPaatosVaihe: cleanupHyvaksymisPaatosVaiheTimestamps(hyvaksymisPaatosVaihe) as Projekti["hyvaksymisPaatosVaihe"],
    }),
    julkaisunSiivoaja: ({ hyvaksymisPaatosVaiheJulkaisu }) => ({
      hyvaksymisPaatosVaiheJulkaisu: cleanupHyvaksymisPaatosVaiheTimestamps(
        hyvaksymisPaatosVaiheJulkaisu
      ) as Projekti["hyvaksymisPaatosVaiheJulkaisu"],
    }),
    julkaisuJulkinenSiivoaja: ({ hyvaksymisPaatosVaihe }) => ({
      hyvaksymisPaatosVaihe: cleanupHyvaksymisPaatosVaiheJulkaisuJulkinenTimestamps(hyvaksymisPaatosVaihe),
    }),
  },
  [UudelleelleenkuulutettavaVaihe.JATKOPAATOS_1]: {
    initialStatus: Status.JATKOPAATOS_1,
    status: Status.JATKOPAATOS_1,
    publicStatus: Status.JATKOPAATOS_1,
    tilasiirtymaTyyppi: TilasiirtymaTyyppi.JATKOPAATOS_1,
    julkaisuKey: "jatkoPaatos1VaiheJulkaisu",
    luonnosKey: "jatkoPaatos1Vaihe",
    luonnoksenSiivoaja: ({ jatkoPaatos1Vaihe }) => ({
      jatkoPaatos1Vaihe: cleanupHyvaksymisPaatosVaiheTimestamps(jatkoPaatos1Vaihe) as Projekti["jatkoPaatos1Vaihe"],
    }),
    julkaisunSiivoaja: ({ jatkoPaatos1VaiheJulkaisu }) => ({
      jatkoPaatos1VaiheJulkaisu: cleanupHyvaksymisPaatosVaiheTimestamps(jatkoPaatos1VaiheJulkaisu) as Projekti["jatkoPaatos1VaiheJulkaisu"],
    }),
    julkaisuJulkinenSiivoaja: ({ jatkoPaatos1Vaihe }) => ({
      jatkoPaatos1Vaihe: cleanupHyvaksymisPaatosVaiheJulkaisuJulkinenTimestamps(jatkoPaatos1Vaihe),
    }),
  },
  [UudelleelleenkuulutettavaVaihe.JATKOPAATOS_2]: {
    initialStatus: Status.JATKOPAATOS_2,
    status: Status.JATKOPAATOS_2,
    publicStatus: Status.JATKOPAATOS_2,
    tilasiirtymaTyyppi: TilasiirtymaTyyppi.JATKOPAATOS_2,
    julkaisuKey: "jatkoPaatos2VaiheJulkaisu",
    luonnosKey: "jatkoPaatos2Vaihe",
    luonnoksenSiivoaja: ({ jatkoPaatos2Vaihe }) => ({
      jatkoPaatos2Vaihe: cleanupHyvaksymisPaatosVaiheTimestamps(jatkoPaatos2Vaihe) as Projekti["jatkoPaatos2Vaihe"],
    }),
    julkaisunSiivoaja: ({ jatkoPaatos2VaiheJulkaisu }) => ({
      jatkoPaatos2VaiheJulkaisu: cleanupHyvaksymisPaatosVaiheTimestamps(jatkoPaatos2VaiheJulkaisu) as Projekti["jatkoPaatos2VaiheJulkaisu"],
    }),
    julkaisuJulkinenSiivoaja: ({ jatkoPaatos2Vaihe }) => ({
      jatkoPaatos2Vaihe: cleanupHyvaksymisPaatosVaiheJulkaisuJulkinenTimestamps(jatkoPaatos2Vaihe),
    }),
  },
};

export async function testUudelleenkuulutus(
  oid: string,
  siirtymaTyyppi: UudelleelleenkuulutettavaVaihe,
  projektiPaallikko: ProjektiKayttaja,
  muokkaaja: NykyinenKayttaja,
  userFixture: UserFixture,
  kuulutusPaiva: string
): Promise<void> {
  const {
    status,
    tilasiirtymaTyyppi,
    julkaisuKey,
    luonnoksenSiivoaja,
    luonnosKey,
    julkaisuJulkinenSiivoaja,
    julkaisunSiivoaja,
    initialStatus,
    publicStatus,
  } = uudelleenKuulutusTiedotVaiheelle[siirtymaTyyppi];

  userFixture.loginAsAdmin();
  await asetaUudelleenkuulutettavaksi(oid, initialStatus, julkaisuKey, tilasiirtymaTyyppi);
  userFixture.logout();

  await testPublicAccessToProjekti(
    oid,
    publicStatus,
    userFixture,
    `${luonnosKey}JulkinenAfterAsetaUudelleenkuulutettavaksi`,
    julkaisuJulkinenSiivoaja
  );

  userFixture.loginAs(muokkaaja);
  await talletaSyytUudelleenkuulutukselleJaLahetaHyvaksyttavaksi(
    oid,
    status,
    tilasiirtymaTyyppi,
    luonnosKey,
    luonnoksenSiivoaja,
    kuulutusPaiva
  );
  userFixture.logout();

  userFixture.loginAsProjektiKayttaja(projektiPaallikko);
  await hyvaksyUudelleenkuulutus(oid, status, julkaisuKey, tilasiirtymaTyyppi, julkaisunSiivoaja);

  await tarkistaProjektinTiedot(oid, status, luonnoksenSiivoaja, julkaisunSiivoaja);

  userFixture.logout();

  asetaAika(kuulutusPaiva);
  await testPublicAccessToProjekti(
    oid,
    publicStatus,
    userFixture,
    `${luonnosKey}JulkinenAfterHyvaksyUudelleenKuulutus`,
    julkaisuJulkinenSiivoaja
  );
}

async function asetaUudelleenkuulutettavaksi(
  oid: string,
  initialStatus: Status,
  julkaisuKey: JulkaisuKey,
  tilasiirtymaTyyppi: TilasiirtymaTyyppi
) {
  const projektiUudelleenkuulutettavaksi = await loadProjektiFromDatabase(oid, initialStatus);
  expect(projektiUudelleenkuulutettavaksi[julkaisuKey]).to.be.an("object");
  expect(projektiUudelleenkuulutettavaksi[julkaisuKey]?.tila).to.eq(KuulutusJulkaisuTila.HYVAKSYTTY);

  await api.siirraTila({ oid, tyyppi: tilasiirtymaTyyppi, toiminto: TilasiirtymaToiminto.UUDELLEENKUULUTA });
}

async function talletaSyytUudelleenkuulutukselleJaLahetaHyvaksyttavaksi(
  oid: string,
  status: Status,
  tilasiirtymaTyyppi: TilasiirtymaTyyppi,
  luonnosKey: LuonnosKey,
  luonnoksenSiivoaja: ProjektinSiivoaja,
  kuulutusPaiva: string
) {
  const projekti = await loadProjektiFromDatabase(oid, status);
  expectToMatchSnapshot("testVaiheAfterSiirraUudelleenkuulutettavaksi", luonnoksenSiivoaja(projekti));

  await api.tallennaProjekti({
    oid,
    versio: projekti.versio,
    [luonnosKey]: {
      kuulutusPaiva,
      kuulutusVaihePaattyyPaiva: dateToString(parseDate(kuulutusPaiva).add(1, "month")),
      uudelleenKuulutus: {
        selosteKuulutukselle: { SUOMI: "Kuulutetaan uudelleen kuulutusteksti ..." },
        selosteLahetekirjeeseen: { SUOMI: "Kuulutetaan uudelleen l√§hetekirjeteksti ..." },
      },
    },
  });

  await api.siirraTila({ oid, toiminto: TilasiirtymaToiminto.LAHETA_HYVAKSYTTAVAKSI, tyyppi: tilasiirtymaTyyppi });
}

async function hyvaksyUudelleenkuulutus(
  oid: string,
  status: Status,
  julkaisuKey: JulkaisuKey,
  tilasiirtymaTyyppi: TilasiirtymaTyyppi,
  julkaisunSiivoaja: ProjektinSiivoaja
) {
  const projekti = await loadProjektiFromDatabase(oid, status);
  expect(projekti[julkaisuKey]).to.be.an("object");
  expect(projekti[julkaisuKey]?.tila).to.eq(KuulutusJulkaisuTila.ODOTTAA_HYVAKSYNTAA);

  expectToMatchSnapshot("testJulkaisuAfterUudelleenkuulutus", julkaisunSiivoaja(projekti));

  await api.siirraTila({ oid, tyyppi: tilasiirtymaTyyppi, toiminto: TilasiirtymaToiminto.HYVAKSY });
}

async function tarkistaProjektinTiedot(
  oid: string,
  status: Status,
  luonnoksenSiivoaja: ProjektinSiivoaja,
  julkaisunSiivoaja: ProjektinSiivoaja
) {
  const projekti = await loadProjektiFromDatabase(oid, status);
  expectToMatchSnapshot("testJulkaisuAfterUudelleenkuulutus", julkaisunSiivoaja(projekti));
  expectToMatchSnapshot("testLuonnosAfterUudelleenkuulutus", luonnoksenSiivoaja(projekti));
}
