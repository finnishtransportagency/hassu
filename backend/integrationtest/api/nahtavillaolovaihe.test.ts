import { describe, it } from "mocha";
import * as sinon from "sinon";
import {
  Aineisto,
  AineistoInput,
  AineistoTila,
  Kieli,
  Status,
  TilasiirtymaToiminto,
  TilasiirtymaTyyppi,
} from "hassu-common/graphql/apiModel";
import { UserFixture } from "../../test/fixture/userFixture";
import { userService } from "../../src/user";
import { cleanupAnyProjektiData } from "./testFixtureRecorder";
import {
  defaultMocks,
  expectToMatchSnapshot,
  mockSaveProjektiToVelho,
  removeTiedosto,
  takePublicS3Snapshot,
  takeYllapitoS3Snapshot,
} from "./testUtil/util";
import { asetaAika, findProjektiPaallikko, tallennaEULogo } from "./testUtil/tests";
import { assertIsDefined } from "../../src/util/assertions";
import { api } from "./apiClient";
import { ProjektiPaths } from "../../src/files/ProjektiPath";
import { createSaameProjektiToVaihe } from "./testUtil/saameUtil";
import { tilaHandler } from "../../src/handler/tila/tilaHandler";
import { testUudelleenkuulutus, UudelleelleenkuulutettavaVaihe } from "./testUtil/uudelleenkuulutus";

describe("Nähtävilläolovaihe", () => {
  const userFixture = new UserFixture(userService);
  const { eventSqsClientMock } = defaultMocks();

  before(async () => {
    mockSaveProjektiToVelho();
  });

  afterEach(() => {
    userFixture.logout();
    sinon.reset();
  });

  after(() => {
    sinon.restore();
  });

  it("suorita nähtävilläolovaihe saamen kielellä onnistuneesti", async function () {
    const dbProjekti = await createSaameProjektiToVaihe(Status.NAHTAVILLAOLO);
    const { oid } = dbProjekti;

    userFixture.loginAs(UserFixture.mattiMeikalainen);
    let p = await api.lataaProjekti(oid);
    const nahtavillaoloVaihe = p.nahtavillaoloVaihe;
    assertIsDefined(nahtavillaoloVaihe);

    asetaAika("2040-01-01");

    // Lataa kuulutus- ja ilmoitustiedostot palveluun. Käytetään olemassa olevaa testitiedostoa, vaikkei se pdf olekaan
    const uploadedIlmoitus = await tallennaEULogo("saameilmoitus.pdf");
    const uploadedKuulutus = await tallennaEULogo("saamekuulutus.pdf");
    const aineistoNahtavillaVanha = dbProjekti.nahtavillaoloVaihe?.aineistoNahtavilla;
    await api.tallennaProjekti({
      oid,
      versio: p.versio,
      nahtavillaoloVaihe: {
        ...nahtavillaoloVaihe,
        aineistoNahtavilla: (
          (aineistoNahtavillaVanha
            ? aineistoNahtavillaVanha.map((item) => removeTiedosto(item) as Omit<Aineisto, "tiedosto">)
            : []) as AineistoInput[]
        ).concat([
          {
            kategoriaId: "FOO",
            nimi: "foo.pdf",
            dokumenttiOid: "1.2.246.578.5.100.2147637429.4251089044",
            tila: AineistoTila.ODOTTAA_TUONTIA,
            uuid: "jotain",
          },
        ]),
        nahtavillaoloSaamePDFt: {
          POHJOISSAAME: { kuulutusPDFPath: uploadedKuulutus, kuulutusIlmoitusPDFPath: uploadedIlmoitus },
        },
      },
    });
    p = await api.lataaProjekti(oid);
    expectToMatchSnapshot(
      "Nähtävilläolovaihe saamenkielisellä kuulutuksella ja ilmoituksella",
      cleanupAnyProjektiData(p.nahtavillaoloVaihe?.nahtavillaoloSaamePDFt || {})
    );
    await takeYllapitoS3Snapshot(
      oid,
      "Nähtävilläolovaihe saamenkielisellä kuulutuksella ja ilmoituksella",
      ProjektiPaths.PATH_NAHTAVILLAOLO
    );
    await eventSqsClientMock.processQueue();

    //
    // Hyväksyntä
    //
    userFixture.loginAs(UserFixture.mattiMeikalainen);
    await tilaHandler.siirraTila({
      oid,
      toiminto: TilasiirtymaToiminto.LAHETA_HYVAKSYTTAVAKSI,
      tyyppi: TilasiirtymaTyyppi.NAHTAVILLAOLO,
    });
    userFixture.loginAs(UserFixture.hassuATunnus1);
    await tilaHandler.siirraTila({
      oid,
      toiminto: TilasiirtymaToiminto.HYVAKSY,
      tyyppi: TilasiirtymaTyyppi.NAHTAVILLAOLO,
    });
    await eventSqsClientMock.processQueue();

    p = await api.lataaProjekti(oid);
    expectToMatchSnapshot(
      "Nähtävilläolovaihe saamenkielisellä kuulutuksella ja ilmoituksella",
      cleanupAnyProjektiData(p.nahtavillaoloVaiheJulkaisu?.nahtavillaoloSaamePDFt || {})
    );
    await takeYllapitoS3Snapshot(
      oid,
      "Nähtävilläolovaihejulkaisu saamenkielisellä kuulutuksella ja ilmoituksella ylläpidossa",
      ProjektiPaths.PATH_NAHTAVILLAOLO
    );
    const julkinenProjekti = await api.lataaProjektiJulkinen(oid, Kieli.SUOMI);
    expectToMatchSnapshot(
      "Julkisen nähtävilläolovaiheen saamenkieliset PDFt",
      cleanupAnyProjektiData(julkinenProjekti.nahtavillaoloVaihe?.nahtavillaoloSaamePDFt || {})
    );
    await takePublicS3Snapshot(
      oid,
      "Nähtävilläolovaihejulkaisu saamenkielisellä kuulutuksella ja ilmoituksella kansalaisille",
      ProjektiPaths.PATH_NAHTAVILLAOLO
    );

    //
    // Uudelleenkuulutus
    //
    const projektiPaallikko = findProjektiPaallikko(p);
    await testUudelleenkuulutus(
      oid,
      UudelleelleenkuulutettavaVaihe.NAHTAVILLAOLO,
      projektiPaallikko,
      UserFixture.mattiMeikalainen,
      userFixture,
      "2040-06-01",
      "nahtavillaoloSaamePDFt"
    );
    await eventSqsClientMock.processQueue();
  });
});
