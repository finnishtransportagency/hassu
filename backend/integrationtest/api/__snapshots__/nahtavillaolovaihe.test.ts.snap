// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nähtävilläolovaihe suorita nähtävilläolovaihe saamen kielellä onnistuneesti 1`] = `
Object {
  "description": "Nähtävilläolovaihe saamenkielisellä kuulutuksella ja ilmoituksella",
  "obj": Object {
    "POHJOISSAAME": Object {
      "__typename": "KuulutusSaamePDF",
      "kuulutusIlmoitusPDF": Object {
        "__typename": "LadattuTiedosto",
        "nimi": "saameilmoitus.pdf",
        "tiedosto": "/yllapito/tiedostot/projekti/3/nahtavillaolo/1/saameilmoitus.pdf",
        "tuotu": "2020-01-01T00:00:00+02:00",
      },
      "kuulutusPDF": Object {
        "__typename": "LadattuTiedosto",
        "nimi": "saamekuulutus.pdf",
        "tiedosto": "/yllapito/tiedostot/projekti/3/nahtavillaolo/1/saamekuulutus.pdf",
        "tuotu": "2020-01-01T00:00:00+02:00",
      },
    },
    "__typename": "KuulutusSaamePDFt",
  },
}
`;

exports[`Nähtävilläolovaihe suorita nähtävilläolovaihe saamen kielellä onnistuneesti 2`] = `
Object {
  "yllapito S3 files Nähtävilläolovaihe saamenkielisellä kuulutuksella ja ilmoituksella": Object {
    "/1/saameilmoitus.pdf": FileMetadata {
      "ContentDisposition": undefined,
      "ContentType": "image/png",
      "checksum": "ABC123",
    },
    "/1/saamekuulutus.pdf": FileMetadata {
      "ContentDisposition": undefined,
      "ContentType": "image/png",
      "checksum": "ABC123",
    },
  },
}
`;

exports[`Nähtävilläolovaihe suorita nähtävilläolovaihe saamen kielellä onnistuneesti 3`] = `
Object {
  "description": "Nähtävilläolovaihe saamenkielisellä kuulutuksella ja ilmoituksella",
  "obj": Object {
    "POHJOISSAAME": Object {
      "__typename": "KuulutusSaamePDF",
      "kuulutusIlmoitusPDF": Object {
        "__typename": "LadattuTiedosto",
        "nimi": "saameilmoitus.pdf",
        "tiedosto": "/yllapito/tiedostot/projekti/3/nahtavillaolo/1/saameilmoitus.pdf",
        "tuotu": "2020-01-01T00:00:00+02:00",
      },
      "kuulutusPDF": Object {
        "__typename": "LadattuTiedosto",
        "nimi": "saamekuulutus.pdf",
        "tiedosto": "/yllapito/tiedostot/projekti/3/nahtavillaolo/1/saamekuulutus.pdf",
        "tuotu": "2020-01-01T00:00:00+02:00",
      },
    },
    "__typename": "KuulutusSaamePDFt",
  },
}
`;

exports[`Nähtävilläolovaihe suorita nähtävilläolovaihe saamen kielellä onnistuneesti 4`] = `
Object {
  "yllapito S3 files Nähtävilläolovaihejulkaisu saamenkielisellä kuulutuksella ja ilmoituksella ylläpidossa": Object {
    "/1/T414 Kuulutus suunnitelman nahtavillaolo.pdf": FileMetadata {
      "ContentDisposition": "inline; filename*=UTF-8''T414%20Kuulutus%20suunnitelman%20nahtavillaolo.pdf",
      "ContentType": "application/pdf",
      "checksum": "ABC123",
      "publishDate": "2022-06-07T00:00:00+03:00",
    },
    "/1/T414_1 Ilmoitus suunnitelman nahtavillaolo.pdf": FileMetadata {
      "ContentDisposition": "inline; filename*=UTF-8''T414_1%20Ilmoitus%20suunnitelman%20nahtavillaolo.pdf",
      "ContentType": "application/pdf",
      "checksum": "ABC123",
      "publishDate": "2022-06-07T00:00:00+03:00",
    },
    "/1/T415 Ilmoitus kiinteistonomistajat nahtaville asettaminen.pdf": FileMetadata {
      "ContentDisposition": "inline; filename*=UTF-8''T415%20Ilmoitus%20kiinteistonomistajat%20nahtaville%20asettaminen.pdf",
      "ContentType": "application/pdf",
      "checksum": "ABC123",
      "publishDate": "2022-06-07T00:00:00+03:00",
    },
    "/1/TYHJÄ.txt": FileMetadata {
      "ContentDisposition": "inline; filename*=UTF-8''TYHJ%C3%84.txt",
      "ContentType": "text/plain",
      "checksum": "ABC123",
      "fileType": "AINEISTO",
    },
    "/1/saameilmoitus.pdf": FileMetadata {
      "ContentDisposition": undefined,
      "ContentType": "image/png",
      "checksum": "ABC123",
    },
    "/1/saamekuulutus.pdf": FileMetadata {
      "ContentDisposition": undefined,
      "ContentType": "image/png",
      "checksum": "ABC123",
    },
  },
}
`;

exports[`Nähtävilläolovaihe suorita nähtävilläolovaihe saamen kielellä onnistuneesti 5`] = `
Object {
  "description": "Julkisen nähtävilläolovaiheen saamenkieliset PDFt",
  "obj": Object {
    "POHJOISSAAME": Object {
      "__typename": "KuulutusSaamePDF",
      "kuulutusIlmoitusPDF": Object {
        "__typename": "LadattuTiedosto",
        "nimi": "saameilmoitus.pdf",
        "tiedosto": "/tiedostot/suunnitelma/3/nahtavillaolo/nahtavillaolo/1/saameilmoitus.pdf",
        "tuotu": "2020-01-01T00:00:00+02:00",
      },
      "kuulutusPDF": Object {
        "__typename": "LadattuTiedosto",
        "nimi": "saamekuulutus.pdf",
        "tiedosto": "/tiedostot/suunnitelma/3/nahtavillaolo/nahtavillaolo/1/saamekuulutus.pdf",
        "tuotu": "2020-01-01T00:00:00+02:00",
      },
    },
    "__typename": "KuulutusSaamePDFt",
  },
}
`;

exports[`Nähtävilläolovaihe suorita nähtävilläolovaihe saamen kielellä onnistuneesti 6`] = `
Object {
  "public S3 files Nähtävilläolovaihejulkaisu saamenkielisellä kuulutuksella ja ilmoituksella kansalaisille": Object {
    "/T414 Kuulutus suunnitelman nahtavillaolo.pdf": FileMetadata {
      "ContentDisposition": "inline; filename*=UTF-8''T414%20Kuulutus%20suunnitelman%20nahtavillaolo.pdf",
      "ContentType": "application/pdf",
      "checksum": "ABC123",
      "publishDate": "2022-06-07T00:00:00+03:00",
    },
    "/T414_1 Ilmoitus suunnitelman nahtavillaolo.pdf": FileMetadata {
      "ContentDisposition": "inline; filename*=UTF-8''T414_1%20Ilmoitus%20suunnitelman%20nahtavillaolo.pdf",
      "ContentType": "application/pdf",
      "checksum": "ABC123",
      "publishDate": "2022-06-07T00:00:00+03:00",
    },
    "/T415 Ilmoitus kiinteistonomistajat nahtaville asettaminen.pdf": FileMetadata {
      "ContentDisposition": "inline; filename*=UTF-8''T415%20Ilmoitus%20kiinteistonomistajat%20nahtaville%20asettaminen.pdf",
      "ContentType": "application/pdf",
      "checksum": "ABC123",
      "publishDate": "2022-06-07T00:00:00+03:00",
    },
    "/saameilmoitus.pdf": FileMetadata {
      "ContentDisposition": undefined,
      "ContentType": "image/png",
      "checksum": "ABC123",
      "publishDate": "2022-06-07T00:00:00+03:00",
    },
    "/saamekuulutus.pdf": FileMetadata {
      "ContentDisposition": undefined,
      "ContentType": "image/png",
      "checksum": "ABC123",
      "publishDate": "2022-06-07T00:00:00+03:00",
    },
  },
}
`;
